src/version_checker.ts

```diff
@@ -0,0 +1,94 @@
+    const nullResult = { version: new Version('0.0.0'), isCorrect: null };
+    let stdout: string;
+    try {
+        stdout = (await runCmd(command)).stdout;
+    } catch (error) {
+        console.error(`Failed to run "${command}": ${error}`);
+        return nullResult;
+    }
+    const found = stdout.match(Version.regex);
+    if (found === null) return nullResult;
+    const version = new Version(stdout);
+    return {
+        version: version,
+        isCorrect: version.isAtLeast(desiredVersion) && (
+            !sameMajor || version.isCompatibleWith(desiredVersion)
+        )
+    };
+}
+
+/**
+ * Return the version checker that returns the version check result that is correct, or the first
+ * version check result if both or neither are correct.
+ */
+const theBetterOfEither: VersionCheckerCombiner = (check0, check1) => async () => {
+    const zeroth: VersionCheckResult = await check0();
+    if (zeroth.isCorrect) return zeroth;
+    const first: VersionCheckResult = await check1();
+    if (first.isCorrect) return first;
+    return zeroth;
+};
+
+export const javaVersionChecker = basicChecker(config.javaVersion, 'java -version 2>&1', true);
+export const javacVersionChecker = basicChecker(config.javacVersion, 'javac -version 2>&1', true);
+export const python3AliasVersionChecker = basicChecker(config.pythonVersion, 'python3 -V', false);
+export const pythonAliasVersionChecker = basicChecker(config.pythonVersion, 'python -V', false);
+export const python3VersionChecker = theBetterOfEither(
+    python3AliasVersionChecker,
+    pythonAliasVersionChecker
+);
+export const nodeVersionChecker = basicChecker(config.nodeVersion, 'node -v', false);
+export const pylintVersionChecker = basicChecker(config.pylintVersion, 'pip3 show pylint', false);
+export const npmVersionChecker = basicChecker(config.npmVersion, 'npm --version', false);
+export const pnpmVersionChecker = basicChecker(config.pnpmVersion, 'pnpm --version', false);
+export const brewVersionChecker = basicChecker(new Version('0.0.0'), 'brew -v', false);
+export const corepackVersionChecker = basicChecker(new Version('0.0.0'), 'corepack -v', false);
+export const curlVersionChecker = basicChecker(new Version('0.0.0'), 'curl -V', false);
+export const rustVersionChecker = basicChecker(config.rustVersion, 'rustc --version', false);
+export const cmakeVersionChecker = basicChecker(config.rustVersion, 'cmake --version', false);
+export const nvmVersionChecker = basicChecker(new Version('0.0.0'), 'nvm --version', false);
+export const snapVersionChecker = basicChecker(new Version('0.0.0'), 'snap --version', false);
+export const aptGetVersionChecker = basicChecker(new Version('0.0.0'), 'apt-get -v', false);
+export const chocolateyVersionChecker = basicChecker(new Version('0.0.0'), 'choco -v', false);
+export const rtiVersionChecker = async () => {
+    // TODO: Update when #1233 is addressed: https://github.com/lf-lang/lingua-franca/issues/1233
```

Comment:
https://github.com/lf-lang/lingua-franca/issues/1233 is closed...

Reply:
```suggestion
    // TODO: Update when #76 is addressed: https://github.com/lf-lang/reactor-c/issues/76
```